src/VisualStudio/Core/Def/Implementation/ProjectSystem/CPS/IWorkspaceProjectContextFactory.cs

```diff
@@ -22,5 +23,19 @@ internal interface IWorkspaceProjectContextFactory
         /// <param name="hierarchy"><see cref="IVsHierarchy"/> for the project, an be null in deferred project load cases.</param>
         /// <param name="binOutputPath">Initial project binary output path.</param>
         IWorkspaceProjectContext CreateProjectContext(string languageName, string projectDisplayName, string projectFilePath, Guid projectGuid, object hierarchy, string binOutputPath);
+
+        /// <summary>
+        /// Creates and initializes a new Workspace project and returns a <see cref="IWorkspaceProjectContext"/> to lazily initialize the properties and items for the project.
+        /// This method can be invoked on a background thread and doesn't access any members of the given UI <paramref name="hierarchy"/>,
+        /// allowing the UI hierarchy to be published lazily.
+        /// </summary>
+        /// <param name="languageName">Project language.</param>
+        /// <param name="projectDisplayName">Display name for the project.</param>
+        /// <param name="projectFilePath">Full path to the project file for the project.</param>
+        /// <param name="projectGuid">Project guid.</param>
+        /// <param name="hierarchy"><see cref="IVsHierarchy"/> for the project, an be null in deferred project load cases.</param>
+        /// <param name="binOutputPath">Initial project binary output path.</param>
+        /// <param name="errorReporter">Error reporter object.</param>
+        IWorkspaceProjectContext CreateProjectContext(String languageName, String projectDisplayName, String projectFilePath, Guid projectGuid, Object hierarchy, String binOutputPath, ProjectExternalErrorReporter errorReporter);
```

Comment:
Nit: Please use predefined type names (like `int/string/object`, instead of framework type names (like `Int32/String/Object`). You might have the option set wrong at Tools\Options\Text Editor\C#\Code style\General.